GH #576: Make sure framebuffer attachments have  correct usage flag

mark-lunarg opened the pull request.

chrisforbes commented on tests/layer_validation_tests.cpp:
```diff
@@ -2922,6 +2922,112 @@ TEST_F(VkLayerTest, BindMemoryToDestroyedObject) {
+    rpci.pAttachments = &attach_desc;
+    rpci.sType = VK_STRUCTURE_TYPE_RENDER_PASS_CREATE_INFO;
+    VkRenderPass rp;
+    vkCreateRenderPass(m_device->device(), &rpci, NULL, &rp);
```
Error is generated here by your new check -- shouldn't be continuing beyond here.

chrisforbes commented on tests/layer_validation_tests.cpp:
```diff
@@ -2922,6 +2922,112 @@ TEST_F(VkLayerTest, BindMemoryToDestroyedObject) {
+
+    m_errorMonitor->VerifyFound();
+
+    vkDestroyRenderPass(m_device->device(), rp, NULL);
```
Tear these down in reverse order?

chrisforbes commented on tests/layer_validation_tests.cpp:
```diff
@@ -2922,6 +2922,112 @@ TEST_F(VkLayerTest, BindMemoryToDestroyedObject) {
+    rpci.pAttachments = &attach_desc;
+    rpci.sType = VK_STRUCTURE_TYPE_RENDER_PASS_CREATE_INFO;
+    VkRenderPass rp;
+    vkCreateRenderPass(m_device->device(), &rpci, NULL, &rp);
```
Should probably assert that this succeeded. If it didn't, you'll get weird behavior later.

chrisforbes commented on layers/core_validation.cpp:
```diff
@@ -8668,42 +8724,50 @@ VKAPI_ATTR VkResult VKAPI_CALL CreateRenderPass(VkDevice device, const VkRenderP
-                skip_call |= log_msg(dev_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, (VkDebugReportObjectTypeEXT)0, 0,
-                                     __LINE__, DRAWSTATE_INVALID_RENDERPASS, "DS",
-                                     "Pipeline bind point for subpass %d must be VK_PIPELINE_BIND_POINT_GRAPHICS.", i);
+                skip_call |=
```
Not new, but this whole function is awful. Accumulating skip_call conditions after making the call!?

mark-lunarg commented:
Thanks for the feedback @chrisforbes.  Changes incorporated and pushed to Master.  I'll create a new issue for the ugliness in CreateRenderPass.  

`commit 6600a7662f6dac6a8c8622a8305f02625439bf30
Author: Mark Lobodzinski <mark@lunarg.com>
Date:   Thu Jun 16 13:23:02 2016 -0600

    layers: GH576, Validate attachment image usage

    Framebuffer attachment images are created with IMAGE_USAGE flags. These
    flags are now validated against their attachment location in subpass
    description structures. Also added a check for a preserve attachment
    being set to UNUSED and fixed a few typos in the attachment index
    checking stuff.

    Change-Id: Id62d3539c490ce8f381d7ae606402980f11946fa`
